htlcswitch/switch.go

```diff
@@ -451,7 +451,26 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, paymentID uint64,
 		htlc:           htlc,
 	}
 
-	return s.forward(packet)
+	circuit := newPaymentCircuit(&htlc.PaymentHash, packet)
+	actions, err := s.circuits.CommitCircuits(circuit)
+	if err != nil {
+		log.Errorf("unable to commit circuit in switch: %v", err)
+		return err
+	}
+
+	// Drop duplicate packet if it has already been seen.
+	switch {
+	case len(actions.Drops) == 1:
+		return ErrDuplicateAdd
+
+	case len(actions.Fails) == 1:
```

Comment:
What about the logic to propagate the error backwards? I think as is, this'll change the error sent back to mission control, which may result in unexpected pruning behavior.

Reply:
Yes, that was still missing. I mentioned that in the comment above. I hadn't requested re-review yet, but thanks anyway for taking another look.

Back-path has been added.